dev/publish_test_results.py

```diff
@@ -0,0 +1,127 @@
+POSTS_DIR = '_posts'
+
+class TestResultPublisher():
+
+  def __init__(self, options):
+    self.__nightly_version = options.nightly_version
+    self.__githubio_repo_uri = options.githubio_repo_uri
+    self.__repo_name = ''
+    self.__test_results_file = options.test_results_file
+    self.__test_results_html = ''
+
+
+  def __checkout_githubio_repo(self):
+    """Clones the spinnaker.github.io git repo.
+    """
+    check_run_quick('git clone {0}'.format(self.__githubio_repo_uri))
+    self.__repo_name = os.path.basename(self.__githubio_repo_uri)
+    if self.__repo_name.endswith('.git'):
+      self.__repo_name = self.__repo_name.replace('.git', '')
+
+  def __read_test_results(self):
+    """Read the test results into memory.
+    """
+    with open(self.__test_results_file, 'r') as results_file:
+      self.__test_results_html = results_file.read()
+
+  def __format_nightly_post(self):
+    # Initialized with 'front matter' necessary for the post.
+    timestamp = '{:%Y-%m-%d %H:%M:%S}'.format(datetime.datetime.now())
+    post_lines = [
+      '---',
+      'title: Spinnaker Nightly Build Version {version}'.format(version=self.__nightly_version),
+      'date: {date}'.format(date=timestamp),
+      'categories: nightly-builds',
+      '---',
+      ''
+    ]
+    post_lines.append('# Spinnaker Nightly Build Version {version}'
+                      .format(version=self.__nightly_version))
+    post_lines.append('\n# Test results\n')
+    post_lines.append(self.__test_results_html)
+    post = '\n'.join(post_lines)
+    return post
+
+  def __publish_post(self, post_content):
+    day = '{:%Y-%m-%d}'.format(datetime.datetime.now())
+    post_name = '{day}-{version}-nightly.md'.format(day=day, version=self.__nightly_version)
+    post_path = os.path.join(self.__repo_name, POSTS_DIR, post_name)
+    # Path to post file relative to the git root.
+    post_rel_path = os.path.join(POSTS_DIR, post_name)
+    with open(post_path, 'w') as post_file:
+      post_file.write(post_content)
+
+    check_run_quick('git -C {0} add {1}'.format(self.__repo_name, post_rel_path))
+    message = 'Nightly Build post for version {0}'.format(self.__nightly_version)
+    check_run_quick('git -C {0} commit -m "{1}"'.format(self.__repo_name, message))
+    check_run_quick('git -C {0} push origin master'.format(self.__repo_name))
+
+  def publish_nightly_post(self):
+    """Creates and publishes Nightly Build post for the spinnaker.github.io site.
+
+    'Publishing' in this case means creating a new file, 'git add'ing it to the
+    local git repository for spinnaker.github.io, and then pushing a commit
+    to origin/master.
+
+    A private key that has access to --githubio_repo needs added
+    to a running ssh-agent on the machine this script will run on:
+
+    > <copy or rsync the key to the vm>
+    > eval `ssh-agent`
+    > ssh-add ~/.ssh/<key with access to github repos>
+    """
+    self.__checkout_githubio_repo()
+    self.__read_test_results()
```

Comment:
This implies either a build failure or a catastrophic test (failure to launch), either way these are problems before here. We can report that a different way, i.e. Jenkins build status, etc.